WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -298,6 +303,10 @@ class PostListViewController: AbstractPostListViewController, UIViewControllerRe
     }
 
     func showCompactOrDefault() {
+        guard isViewOnScreen() else {
```

Comment:
How come this is necessary now? Is it because of the call for this method being moved to viewWillAppear?

Reply:
Should we just move it to `viewDidAppear`?

Reply:
This method is called when `isCompact` is changed (and it's changed in a few different places).

Moving it to `viewDidAppear` doesn't fix the issue, it will require a bigger refactor. However, after testing again I saw that this was causing some problems with icons and labels not being properly set.

I changed this method in 38c10d0768f9dd4351f7f799a951f54149fbfefe

Now, it will be executed, but it will only reload the tableView if the view is on screen.